src/main/java/com/iexec/sms/admin/ApiKeyRequestFilter.java

```diff
@@ -43,40 +44,44 @@ public class ApiKeyRequestFilter extends GenericFilterBean {
     private static final String API_KEY_HEADER_NAME = "X-API-KEY"; //Name of header in which api key is expected
     private final String apiKey; //The filter API Key
 
-    private final boolean isEnabled;
+    private final boolean isSensitiveApiEnabled;
 
     public ApiKeyRequestFilter(String apiKey) {
-        if (null != apiKey && !apiKey.isBlank()) {
-            this.apiKey = apiKey;
-            this.isEnabled = true;
-        } else {
+        if (StringUtils.isBlank(apiKey)) {
+            log.warn("No API key has been set. Sensitive API is therefore disabled.");
             this.apiKey = null;
-            this.isEnabled = false;
-            log.warn("API Key filter is not enabled");
+            this.isSensitiveApiEnabled = false;
+            return;
         }
+        this.apiKey = apiKey;
+        this.isSensitiveApiEnabled = true;
     }
 
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
             throws IOException, ServletException {
+        if (!this.isSensitiveApiEnabled) {
+            sendResponseWithStatusAndMessage(response, "This endpoint is disabled.", HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
 
-        if (this.isEnabled) {
-            HttpServletRequest req = (HttpServletRequest) request;
-
-            String key = req.getHeader(API_KEY_HEADER_NAME);
-            if (!this.apiKey.equalsIgnoreCase(key)) {
-                HttpServletResponse resp = (HttpServletResponse) response;
-                String error = "You are not authorized to access this endpoint";
+        final HttpServletRequest req = (HttpServletRequest) request;
 
-                resp.reset();
-                resp.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-                response.setContentLength(error.length());
-                response.getWriter().write(error);
-            } else {
-                chain.doFilter(request, response);
-            }
-        } else {
-            chain.doFilter(request, response);
+        final String key = req.getHeader(API_KEY_HEADER_NAME);
+        if (!this.apiKey.equals(key)) {
+            sendResponseWithStatusAndMessage(response, "You are not authorized to access this endpoint.", HttpServletResponse.SC_UNAUTHORIZED);
+            return;
         }
+
+        chain.doFilter(request, response);
+    }
+
+    private static void sendResponseWithStatusAndMessage(ServletResponse response, String error, int scUnauthorized) throws IOException {
```

Comment:
I don't really like to see Unauthorized here as another status will be Forbidden
```suggestion
    private static void sendResponseWithStatusAndMessage(ServletResponse response, String error, int scStatus) throws IOException {
```